src/components/auth/SignUp.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { View, Text, StyleSheet } from 'react-native';
+import { connect } from 'react-redux';
+import { nameChanged, emailChanged, passwordChanged } from '../../actions';
+import { Button, AppBackgroundView } from '../common';
+import Input from './Input';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'flex-start',
+    backgroundColor: 'black',
+  },
+  facebookButtonContainer: {
+    height: 100,
+    justifyContent: 'center',
+    backgroundColor: 'navy',
+  },
+  facebookButtonStyle: {
+    marginHorizontal: 25,
+    height: 50,
+  },
+  inputContainer: {
+    flex: 1,
+    marginTop: 10,
+  },
+  signUpButtonContainer: {
+    flex: 1,
+    justifyContent: 'flex-end',
+    marginBottom: 25,
+  },
+  signUpButtonStyle: {
+    marginHorizontal: 20,
+    height: 50,
+  },
+  ToSContainer: {
+    alignItems: 'center',
+    marginTop: 15,
+  },
+  ToSTextStyle: {
+    color: 'white',
+    textAlign: 'center',
+  },
+  ToSLinkStyle: {
+    color: '#007fff',
+  },
+});
+
+class SignUp extends Component {
+  constructor() {
+    super();
+
+    this.onNameChange = this.onNameChange.bind(this);
+    this.onEmailChange = this.onEmailChange.bind(this);
+    this.onPasswordChange = this.onPasswordChange.bind(this);
+  }
+  onNameChange(text) {
+    this.props.nameChanged(text);
+  }
+  onEmailChange(text) {
+    this.props.emailChanged(text);
+  }
+  onPasswordChange(text) {
+    this.props.passwordChanged(text);
+  }
+  render() {
+    return (
+      <AppBackgroundView style={styles.container}>
+        <View style={styles.facebookButtonContainer}>
+          <View style={styles.facebookButtonStyle}>
+            <Button backgroundColor="#007fff" fontColor="white">
+              SIGN UP WITH FACEBOOK
+            </Button>
+          </View>
+        </View>
+        <View style={styles.inputContainer}>
+          <Input
+            label="NAME"
+            placeholder="JOHN DOE"
+            value={this.props.name}
+            onChangeText={this.onNameChange}
+          />
+          <Input
+            label="EMAIL"
+            placeholder="HELLO@WOLRD.COM"
+            value={this.props.email}
```

Comment:
i guess what i said above does apply here. if you modify the value in the login component, that will show up here in the SignUp component, which is unusual behavior.